tools/distpartitioning/convert_partition.py

```diff
@@ -0,0 +1,448 @@
+import os
+import json
+import time
+import argparse
+import numpy as np
+import dgl
+import torch as th
+import pyarrow
+import pandas as pd
+from pyarrow import csv
+
+def read_feats(file_name):
+    attrs = csv.read_csv(file_name, read_options=pyarrow.csv.ReadOptions(autogenerate_column_names=True),
+                         parse_options=pyarrow.csv.ParseOptions(delimiter=' '))
+    num_cols = len(attrs.columns)
+    return np.stack([attrs.columns[i].to_numpy() for i in range(num_cols)], 1)
+
+def create_dgl_object(input_dir, graph_name, num_parts, num_node_weights, node_attr_dtype, \
```

Comment:
i think this function should not involve I/O. that is, it doesn't read data from files and save data to files. All data can be passed to it as input arguments and all data are returned as values.